frontend/packages/pipelines-plugin/integration-tests/support/pages/pipelines/pipelineBuilder-page.ts

```diff
@@ -61,16 +71,65 @@ export const pipelineBuilderPage = {
     cy.get(pipelineBuilderPO.formView.quickSearch).type(taskName);
     cy.byTestID('task-cta').click();
   },
-  selectTask: (taskName: string = 'kn') => {
-    cy.get('body').then(($body) => {
-      if ($body.text().includes('Unable to locate any tasks.')) {
-        cy.reload();
+  clickAddTask: (tries: number = 6, polling: number = 20000) => {
+    if (tries === 0) {
+      return;
+    }
+
+    cy.get('body').then(($ele) => {
+      if ($ele.find(pipelineBuilderPO.formView.taskDropdown).length === 0) {
+        createForm.clickCancel();
+        perspective.switchTo(switchPerspective.Developer);
+        projectNameSpace.selectOrCreateProject(Cypress.env('NAMESPACE'));
+        navigateTo(devNavigationMenu.Pipelines);
+        //  Due to dependency cycle error, Adding below code related to pipelinesPage.clickOnCreatePipeline();
+        detailsPage.titleShouldContain(pageTitle.Pipelines);
+        cy.get('body').then(($body) => {
+          if ($body.find('[data-test-id= "dropdown-button"]').length > 0) {
+            cy.byLegacyTestID('dropdown-button').click();
+            cy.get(pipelineBuilderPO.pipeline).click();
+          } else {
+            cy.get(pipelinesPO.createPipeline).click();
+          }
+        });
+
+        pipelineBuilderPage.enterPipelineName(Cypress.env('PIPELINE_NAME'));
+        cy.byTestID('loading-indicator').should('not.exist');
+        // Element was not found. Wait and try again.
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        cy.wait(polling);
+        if ($ele.find('[aria-label="Danger Alert"]').length > 0) {
+          pipelineBuilderPage.clickAddTask(tries - 1, polling);
+        } else {
+          cy.get(pipelineBuilderPO.formView.taskDropdown).click();
+        }
+      } else {
+        cy.get(pipelineBuilderPO.formView.taskDropdown).click();
       }
     });
-    cy.get(pipelineBuilderPO.formView.taskDropdown).click();
+  },
+  selectTask: (taskName: string = 'kn', tries: number = 4, polling: number = 500) => {
+    pipelineBuilderPage.clickAddTask(tries, polling);
+    cy.request(
+      `api/kubernetes/apis/tekton.dev/v1beta1/namespaces/${Cypress.env(
+        'NAMESPACE',
+      )}/tasks?limit=250`,
+    ).then((resp) => {
+      expect(resp.status).toEqual(200);
+    });
+    cy.request('api/kubernetes/apis/tekton.dev/v1beta1/clustertasks?limit=250').then((resp) => {
+      expect(resp.status).toEqual(200);
+      expect(resp.headers).toContain({
+        'content-encoding': 'gzip',
+        'transfer-encoding': 'chunked',
+      });
+    });
```

Comment:
What's the point of these requests if all it does it expect the URL to not fail?
Also why limit 250 instead of just 1?

Reply:
It waits to load the tasks and cluster tasks, so there are limited chances to failures

Reply:
This seems kind of arbitrary as a means to add some sort of delay.
How does this help to select the task? Is there not a true deterministic way to wait for whatever it is you need?

Reply:
Its a double check to make sure tasks are loaded, I can remove those lines

Reply:
What confuses me is that it's an arbitrary network call and there's no assertions on the data. This is why I'm trying to understand why you're doing them.

Reply:
Some times tasks are not loaded, so if we

> @makambalaji you said the following in regards to the `wait` that was previously present:
> 
> > Its only for pipelines operator, even though it is present in generic functions and also if pipelines operator is already installed this wait time is not applicable.
> > Even if status is displaying as expected, still the pipelines tab doesn't display immediately and also tasks doesn't appear in pipelines page. So IMO this is recommended wait.
> 
> Now you've removed all the waits. Is this still an issue or did you put in other appropriate checks instead? If you're waiting on the tab to be present then an assertion to get the tab with a timeout should be added instead of arbitrary waits.

cy.request() verifies and [wait](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/integration-tests/cypress.json#L10) for the request to complete. Executed multiple times and verified. Hope this is the better solution instead waits.

Reply:
If pipeline operator is properly installed, this check is not required. So the above cy.request() should fix this issue

Reply:
This code is no longer required, removed from this pr and replaced with the UI checks : https://github.com/openshift/console/pull/10329/files#diff-7b7c13976a5803927e67a42d216a93d88575b5eb3288214eb35fde825853a4aeR60-R78